crates/data_components/src/mssql.rs

```diff
@@ -0,0 +1,592 @@
+        ColumnType::Image | ColumnType::BigVarBin | ColumnType::BigBinary => DataType::Binary,
+        ColumnType::Guid
+        | ColumnType::BigVarChar
+        | ColumnType::BigChar
+        | ColumnType::NVarchar
+        | ColumnType::NChar
+        | ColumnType::Xml
+        | ColumnType::Udt
+        | ColumnType::Text
+        | ColumnType::NText
+        | ColumnType::SSVariant => DataType::Utf8,
+        ColumnType::Bit | ColumnType::Bitn => DataType::Boolean,
+    }
+}
+
+fn map_type_name_to_column_type(data_type: &str) -> Result<ColumnType> {
+    // https://github.com/prisma/tiberius/blob/51f0cbb3e430db74ba0ea4830b236e89f1b1e03f/src/tds/codec/token/token_col_metadata.rs#L26
+    let column_type = match data_type.to_lowercase().as_str() {
+        "int" => ColumnType::Int4,
+        "bigint" => ColumnType::Int8,
+        "smallint" => ColumnType::Int2,
+        "tinyint" => ColumnType::Int1,
+        "float" => ColumnType::Float8,
+        "real" => ColumnType::Float4,
+        "decimal" | "numeric" => ColumnType::Decimaln,
+        "char" | "varchar" | "text" => ColumnType::BigVarChar,
+        "nchar" | "nvarchar" | "ntext" => ColumnType::NVarchar,
+        "uniqueidentifier" => ColumnType::Guid,
+        "binary" => ColumnType::BigBinary,
+        "varbinary" | "image" => ColumnType::BigVarBin,
+        "xml" => ColumnType::Xml,
+        "money" => ColumnType::Money,
+        "date" => ColumnType::Daten,
+        "time" => ColumnType::Timen,
+        "datetime" => ColumnType::Datetime,
+        "smalldatetime" => ColumnType::Datetime4,
+        "datetime2" => ColumnType::Datetime2,
+        "datetimeoffset" => ColumnType::DatetimeOffsetn,
+        "bit" => ColumnType::Bit,
+        other => {
+            return Err(Error::UnsupportedType {
+                data_type: other.to_string(),
+            })
+        }
+    };
+
+    Ok(column_type)
+}
+
+#[async_trait]
+impl TableProvider for SqlServerTableProvider {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        Arc::clone(&self.schema)
+    }
+
+    fn table_type(&self) -> TableType {
+        TableType::Base
+    }
+
+    fn supports_filters_pushdown(
+        &self,
+        filters: &[&Expr],
+    ) -> std::result::Result<Vec<TableProviderFilterPushDown>, DataFusionError> {
+        Ok(vec![
+            TableProviderFilterPushDown::Unsupported;
+            filters.len()
+        ])
+    }
+
+    async fn scan(
+        &self,
+        state: &dyn Session,
+        projection: Option<&Vec<usize>>,
+        filters: &[Expr],
+        limit: Option<usize>,
+    ) -> datafusion::error::Result<Arc<dyn ExecutionPlan>> {
+        let sql = format!("SELECT * from {dataset}", dataset = self.table);
+        let records = self.query_arrow(&sql, Arc::clone(&self.schema)).await?;
+
+        let table = MemTable::try_new(Arc::clone(&self.schema), vec![records])?;
+        table.scan(state, projection, filters, limit).await
```

Comment:
The actual query shouldn't be done in the `scan` method. We need to implement an ExecutionPlan that defers the actual query to when .execute is called.

Reply:
This can come in a follow-on PR

Reply:
Confirm I'll do this as a next PR